ta/pkcs11/src/entry.c

```diff
@@ -136,17 +154,28 @@ TEE_Result TA_InvokeCommandEntryPoint(void *tee_session __unused, uint32_t cmd,
 
 	switch (cmd) {
 	case PKCS11_CMD_PING:
-		res = entry_ping(ctrl, p1_in, p2_out);
+		rc = entry_ping(ctrl, p1_in, p2_out);
 		break;
 
 	default:
 		EMSG("Command 0x%"PRIx32" is not supported", cmd);
 		return TEE_ERROR_NOT_SUPPORTED;
 	}
 
-	/* Currently no output data stored in output param#0 */
-	if (TEE_PARAM_TYPE_GET(ptypes, 0) == TEE_PARAM_TYPE_MEMREF_INOUT)
-		ctrl->memref.size = 0;
+	if (ctrl_stores_output_status(ptypes, ctrl)) {
+		TEE_MemMove(ctrl->memref.buffer, &rc, sizeof(uint32_t));
+		ctrl->memref.size = sizeof(uint32_t);
+
+		res = pkcs2tee_noerr(rc);
+
+		DMSG("%s rc 0x%08"PRIx32"/%s",
+		     id2str_ta_cmd(cmd), rc, id2str_rc(rc));
+	} else {
+		res = pkcs2tee_error(rc);
```

Comment:
It seems to complicate things with this flexibility. Either is the ctrl buffer OK for our purpose or we error out early.

Reply:
You mean always using ctrl output data as return value for all commands?
I'm fine. Just need to update some commands... oh! only the PING command is defined. Well, it currently does not use ctrl output data.

+1 to update accordingly the TA API as a former commit in this series?

Reply:
Yes, I hope you're agreeing.

Reply:
Yes, I agree. I just wanted you to confirm in case I misunderstood your suggestion.

Reply:
Note the change will break IBART CI since optee_client/libckteec must be updated accordingly.
I'll create a P-R.

Reply:
> Note the change will break IBART CI since optee_client/libckteec must be updated accordingly.

Note, that I (still) haven't enabled the PKCS related stuff in IBART, but here I suppose you're talking about dependencies between gits.

Reply:
> Note, that I (still) haven't enabled the PKCS related stuff in IBART

ok :).   Then CI won't complain.

Reply:
Well well well...
Factorizing the return status among command make me think I should fully change the way pkcs11 status are reported to client.
Currently, I use param#0 output buffer. It is mostly already used (planned) for command serialized structured data.

I think I should change to use TA vendor specific return code through GPD standard tee result, rather than an output param memref.
TA would return `TEEC_SUCCESS`, `TEEC_ERROR_SHORT_BUFFER` or a specific ID with BIT(31) set.

Reply:
for the record: bad idea to use vendor specific return codes. We must use TEE standard return value.